src/components/manifold-subscription-create/manifold-subscription-create.tsx

```diff
@@ -205,36 +215,45 @@ export class ManifoldSubscriptionCreate {
         },
       });
 
-      this.subscribing = false;
-
       if (error) {
+        this.subscribing = false;
         this.setupIntentError = error.message;
-      } else {
-        this.setupIntentStatus = setupIntent?.status;
-        if (setupIntent?.status === 'succeeded') {
-          const configuredFeatures = Object.keys(this.configuredFeatures).map(key => ({
-            label: key,
-            value: `${this.configuredFeatures[key]}`,
-          }));
-
-          const variables: CreateSubscriptionMutationVariables = {
-            ownerId: undefined,
-            planId: this.planId,
-            configuredFeatures,
-          };
-
-          const res = await this.connection?.graphqlFetch<CreateSubscriptionMutation>({
-            query: createSubscrptionMutation,
-            variables,
-          });
-
-          // eslint-disable-next-line no-console
-          console.log({ res });
+        return;
+      }
+
+      this.setupIntentStatus = setupIntent?.status;
+      if (setupIntent?.status === 'succeeded' && this.connection) {
+        const configuredFeatures = Object.keys(this.configuredFeatures).map(key => ({
+          label: key,
+          value: `${this.configuredFeatures[key]}`,
+        }));
+
+        const variables: CreateSubscriptionMutationVariables = {
+          ownerId: undefined,
+          planId: this.planId,
+          configuredFeatures,
+        };
+
+        const { data, errors } = await this.connection.graphqlFetch<CreateSubscriptionMutation>({
+          query: createSubscrptionMutation,
+          variables,
+        });
+
+        if (errors) {
+          // TODO handle errors
+        }
+
+        if (data) {
+          console.log(data);
+          this.createSuccess.emit({ id: data.createSubscription.data.id });
```

Comment:
Does this do anything to fire the event?

```diff
- async fetchCustomCost(configuredFeatures: FeatureMap) {
+fetchCustomCost: async (configuredFeatures: FeatureMap) => {
```

Reply:
Oh wait sorry; this is in a different function (I misread the diff). I don’t see anything apparently off with this code; seems like it should work

Reply:
Oh maybe it’s the `onSubmit={this.subscribe}`. Does `onSubmit{(evt) => this.subscribe(evt)}` do anything differently?

Reply:
It's was all working correctly. Sam's solution works. I think it was adding the event listener too early before the element was added to the dom